sql/hive/src/main/scala/org/apache/spark/sql/hive/execution/HiveFileFormat.scala

```diff
@@ -142,7 +142,8 @@ class HiveOutputWriter(
 
   override def write(row: InternalRow): Unit = {
     var i = 0
-    while (i < fieldOIs.length) {
+    val fieldOIsLength = fieldOIs.length
+    while (i < fieldOIsLength) {
```

Comment:
I have tried the test 10 times , and the average cost is almost same, I will close the PR, thanks!